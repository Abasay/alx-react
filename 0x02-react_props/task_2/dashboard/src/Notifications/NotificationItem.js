import React from 'react'

export const NotificationItem = ({ type, html, value }) => {
  return (
    <li data-notification-type={type} dangerouslySetInnerHTML={html}>
      {value}
    </li>
  )
}
