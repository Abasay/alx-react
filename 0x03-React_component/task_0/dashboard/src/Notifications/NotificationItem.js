import React from 'react'
import PropTypes, { string } from 'prop-types'

export const NotificationItem = ({ type, html, value }) => {
  return (
    <li data-notification-type={type} dangerouslySetInnerHTML={html}>
      {value}
    </li>
  )
}

NotificationItem.propTypes = {
  html: PropTypes.shape({ __html: string }),
  type: PropTypes.string,
  value: PropTypes.string,
}

NotificationItem.defaultProps = {
  type: 'default',
}
