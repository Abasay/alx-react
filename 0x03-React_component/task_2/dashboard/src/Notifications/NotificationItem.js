import React from 'react'
import PropTypes, { string } from 'prop-types'

export const NotificationItem = ({
  notificationId,
  type,
  html,
  value,
  markAsRead,
}) => {
  return (
    <li
      data-notification-type={type}
      dangerouslySetInnerHTML={html}
      onClick={() => markAsRead(notificationId)}
    >
      {value}
    </li>
  )
}

NotificationItem.propTypes = {
  notificationId: PropTypes.number,
  html: PropTypes.shape({ __html: string }),
  type: PropTypes.string,
  value: PropTypes.string,
  markAsRead: PropTypes.func,
}

NotificationItem.defaultProps = {
  type: 'default',
}
