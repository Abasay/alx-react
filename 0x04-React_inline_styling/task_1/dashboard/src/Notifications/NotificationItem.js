import React from 'react'
import PropTypes, { string } from 'prop-types'

export class NotificationItem extends React.PureComponent {
  constructor(props) {
    super(props)
  }
  render() {
    return (
      <li
        data-notification-type={this.props.type}
        dangerouslySetInnerHTML={this.props.html}
        onClick={() => markAsRead(this.props.notificationId)}
      >
        {this.props.value}
      </li>
    )
  }
}

NotificationItem.propTypes = {
  notificationId: PropTypes.number,
  html: PropTypes.shape({ __html: string }),
  type: PropTypes.string,
  value: PropTypes.string,
  markAsRead: PropTypes.func,
}

NotificationItem.defaultProps = {
  type: 'default',
}
