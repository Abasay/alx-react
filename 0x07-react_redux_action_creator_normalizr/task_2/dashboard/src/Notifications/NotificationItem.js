import React from 'react';
import PropTypes, { string } from 'prop-types';
import { AppContext, markNotificationAsRead } from '../App/AppContext';

export class NotificationItem extends React.PureComponent {
  constructor(props) {
    super(props);
  }
  static contextType = AppContext;
  render() {
    let { markNotificationAsRead } = this.context;
    return (
      <li
        data-notification-type={this.props.type}
        dangerouslySetInnerHTML={this.props.html}
        onClick={() => markNotificationAsRead(this.props.notificationId)}
      >
        {this.props.value}
      </li>
    );
  }
}

NotificationItem.propTypes = {
  notificationId: PropTypes.number,
  html: PropTypes.shape({ __html: string }),
  type: PropTypes.string,
  value: PropTypes.string,
  markNotificationAsRead: PropTypes.func,
};

NotificationItem.defaultProps = {
  type: 'default',
};
