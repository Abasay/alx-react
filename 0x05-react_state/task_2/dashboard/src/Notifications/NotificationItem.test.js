import React from 'react'
import { shallow } from 'enzyme'
import { NotificationItem } from './NotificationItem'
import { screen, render } from '@testing-library/react'

describe('Notification Item test', () => {
  it('should render without fail', () => {
    const wrapper = shallow(<NotificationItem />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the test', () => {
    render(<NotificationItem type='default' value='test' />)
    const txt = screen.getByText('test')
    expect(txt).toBeInTheDocument()
  })

  it('Should render when passed the html prop', () => {
    const htmlProp = { __html: '<u>text</u>' }
    const wrapper = shallow(<NotificationItem html={htmlProp} />)
    const txt = wrapper.find('li')
    expect(txt.html()).toContain('<u>text</u>')
  })
})
