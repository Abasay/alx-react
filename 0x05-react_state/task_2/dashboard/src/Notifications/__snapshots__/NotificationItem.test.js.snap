// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Item test should render without fail 1`] = `ShallowWrapper {}`;
